### ISSUE TITLE
Jenkins failure at step 6

### ISSUE THREAD

**Issue by phila6** [dd010101/vyos-jenkins#34]
When I run step 6, I get Jenkins failure for vyos-1x andvyos-cloud-init .

Any idea why the curl times out? The job runs for about 2-3mins and then fails.

Example error output:

```
Legacy code started this job.  No cause information is available
[Mon Jul 08 21:01:59 UTC 2024] Starting branch indexing...
 > git --version # timeout=10
 > git --version # 'git version 2.39.2'
 > git ls-remote --symref -- https://github.com/vyos/vyos-cloud-init.git # timeout=10
 > git rev-parse --resolve-git-dir /var/lib/jenkins/caches/git-110bd3a0d6497c5a4e2c2934bf58f1b6/.git # timeout=10
Setting origin to https://github.com/vyos/vyos-cloud-init.git
 > git config remote.origin.url https://github.com/vyos/vyos-cloud-init.git # timeout=10
Fetching & pruning origin...
Listing remote references...
 > git config --get remote.origin.url # timeout=10
 > git --version # timeout=10
 > git --version # 'git version 2.39.2'
 > git ls-remote -h -- https://github.com/vyos/vyos-cloud-init.git # timeout=10
Fetching upstream changes from origin
 > git config --get remote.origin.url # timeout=10
 > git fetch --tags --force --progress --prune -- origin +refs/heads/*:refs/remotes/origin/* # timeout=10
ERROR: [Mon Jul 08 21:03:31 UTC 2024] Could not fetch branches from source 61655632-2acc-4cf1-be05-2bd1530809a4
[Mon Jul 08 21:03:31 UTC 2024] Finished branch indexing. Indexing took 1 min 32 sec
FATAL: Failed to recompute children of vyos-cloud-init
hudson.plugins.git.GitException: Command "git fetch --tags --force --progress --prune -- origin +refs/heads/*:refs/remotes/origin/*" returned status code 128:
stdout: 
stderr: remote: Enumerating objects: 40851, done.        
remote: Counting objects:   0% (1/5846)        
remote: Counting objects:   1% (59/5846)        
remote: Counting objects:   2% (117/5846)        
remote: Counting objects:   3% (176/5846)        
remote: Counting objects:   4% (234/5846)        
remote: Counting objects:   5% (293/5846)        
remote: Counting objects:   6% (351/5846)        
remote: Counting objects:   7% (410/5846)        
remote: Counting objects:   8% (468/5846)        
remote: Counting objects:   9% (527/5846)        
remote: Counting objects:  10% (585/5846)        
remote: Counting objects:  11% (644/5846)        
remote: Counting objects:  12% (702/5846)        
remote: Counting objects:  13% (760/5846)        
remote: Counting objects:  14% (819/5846)        
remote: Counting objects:  15% (877/5846)        
remote: Counting objects:  16% (936/5846)        
remote: Counting objects:  17% (994/5846)        
remote: Counting objects:  18% (1053/5846)        
remote: Counting objects:  19% (1111/5846)        
remote: Counting objects:  20% (1170/5846)        
remote: Counting objects:  21% (1228/5846)        
remote: Counting objects:  22% (1287/5846)        
remote: Counting objects:  23% (1345/5846)        
remote: Counting objects:  24% (1404/5846)        
remote: Counting objects:  25% (1462/5846)        
remote: Counting objects:  26% (1520/5846)        
remote: Counting objects:  27% (1579/5846)        
remote: Counting objects:  28% (1637/5846)        
remote: Counting objects:  29% (1696/5846)        
remote: Counting objects:  30% (1754/5846)        
remote: Counting objects:  31% (1813/5846)        
remote: Counting objects:  32% (1871/5846)        
remote: Counting objects:  33% (1930/5846)        
remote: Counting objects:  34% (1988/5846)        
remote: Counting objects:  35% (2047/5846)        
remote: Counting objects:  36% (2105/5846)        
remote: Counting objects:  37% (2164/5846)        
remote: Counting objects:  38% (2222/5846)        
remote: Counting objects:  39% (2280/5846)        
remote: Counting objects:  40% (2339/5846)        
remote: Counting objects:  41% (2397/5846)        
remote: Counting objects:  42% (2456/5846)        
remote: Counting objects:  43% (2514/5846)        
remote: Counting objects:  44% (2573/5846)        
remote: Counting objects:  45% (2631/5846)        
remote: Counting objects:  46% (2690/5846)        
remote: Counting objects:  47% (2748/5846)        
remote: Counting objects:  48% (2807/5846)        
remote: Counting objects:  49% (2865/5846)        
remote: Counting objects:  50% (2923/5846)        
remote: Counting objects:  51% (2982/5846)        
remote: Counting objects:  52% (3040/5846)        
remote: Counting objects:  53% (3099/5846)        
remote: Counting objects:  54% (3157/5846)        
remote: Counting objects:  55% (3216/5846)        
remote: Counting objects:  56% (3274/5846)        
remote: Counting objects:  57% (3333/5846)        
remote: Counting objects:  58% (3391/5846)        
remote: Counting objects:  59% (3450/5846)        
remote: Counting objects:  60% (3508/5846)        
remote: Counting objects:  61% (3567/5846)        
remote: Counting objects:  62% (3625/5846)        
remote: Counting objects:  63% (3683/5846)        
remote: Counting objects:  64% (3742/5846)        
remote: Counting objects:  65% (3800/5846)        
remote: Counting objects:  66% (3859/5846)        
remote: Counting objects:  67% (3917/5846)        
remote: Counting objects:  68% (3976/5846)        
remote: Counting objects:  69% (4034/5846)        
remote: Counting objects:  70% (4093/5846)        
remote: Counting objects:  71% (4151/5846)        
remote: Counting objects:  72% (4210/5846)        
remote: Counting objects:  73% (4268/5846)        
remote: Counting objects:  74% (4327/5846)        
remote: Counting objects:  75% (4385/5846)        
remote: Counting objects:  76% (4443/5846)        
remote: Counting objects:  77% (4502/5846)        
remote: Counting objects:  78% (4560/5846)        
remote: Counting objects:  79% (4619/5846)        
remote: Counting objects:  80% (4677/5846)        
remote: Counting objects:  81% (4736/5846)        
remote: Counting objects:  82% (4794/5846)        
remote: Counting objects:  83% (4853/5846)        
remote: Counting objects:  84% (4911/5846)        
remote: Counting objects:  85% (4970/5846)        
remote: Counting objects:  86% (5028/5846)        
remote: Counting objects:  87% (5087/5846)        
remote: Counting objects:  88% (5145/5846)        
remote: Counting objects:  89% (5203/5846)        
remote: Counting objects:  90% (5262/5846)        
remote: Counting objects:  91% (5320/5846)        
remote: Counting objects:  92% (5379/5846)        
remote: Counting objects:  93% (5437/5846)        
remote: Counting objects:  94% (5496/5846)        
remote: Counting objects:  95% (5554/5846)        
remote: Counting objects:  96% (5613/5846)        
remote: Counting objects:  97% (5671/5846)        
remote: Counting objects:  98% (5730/5846)        
remote: Counting objects:  99% (5788/5846)        
remote: Counting objects: 100% (5846/5846)        
remote: Counting objects: 100% (5846/5846), done.        
remote: Compressing objects:   0% (1/1642)        
remote: Compressing objects:   1% (17/1642)        
remote: Compressing objects:   2% (33/1642)        
remote: Compressing objects:   3% (50/1642)        
remote: Compressing objects:   4% (66/1642)        
remote: Compressing objects:   5% (83/1642)        
remote: Compressing objects:   6% (99/1642)        
remote: Compressing objects:   7% (115/1642)        
remote: Compressing objects:   8% (132/1642)        
remote: Compressing objects:   9% (148/1642)        
remote: Compressing objects:  10% (165/1642)        
remote: Compressing objects:  11% (181/1642)        
remote: Compressing objects:  12% (198/1642)        
remote: Compressing objects:  13% (214/1642)        
remote: Compressing objects:  14% (230/1642)        
remote: Compressing objects:  15% (247/1642)        
remote: Compressing objects:  16% (263/1642)        
remote: Compressing objects:  17% (280/1642)        
remote: Compressing objects:  18% (296/1642)        
remote: Compressing objects:  19% (312/1642)        
remote: Compressing objects:  20% (329/1642)        
remote: Compressing objects:  21% (345/1642)        
remote: Compressing objects:  22% (362/1642)        
remote: Compressing objects:  23% (378/1642)        
remote: Compressing objects:  24% (395/1642)        
remote: Compressing objects:  25% (411/1642)        
remote: Compressing objects:  26% (427/1642)        
remote: Compressing objects:  27% (444/1642)        
remote: Compressing objects:  28% (460/1642)        
remote: Compressing objects:  29% (477/1642)        
remote: Compressing objects:  30% (493/1642)        
remote: Compressing objects:  31% (510/1642)        
remote: Compressing objects:  32% (526/1642)        
remote: Compressing objects:  33% (542/1642)        
remote: Compressing objects:  34% (559/1642)        
remote: Compressing objects:  35% (575/1642)        
remote: Compressing objects:  36% (592/1642)        
remote: Compressing objects:  37% (608/1642)        
remote: Compressing objects:  38% (624/1642)        
remote: Compressing objects:  39% (641/1642)        
remote: Compressing objects:  40% (657/1642)        
remote: Compressing objects:  41% (674/1642)        
remote: Compressing objects:  42% (690/1642)        
remote: Compressing objects:  43% (707/1642)        
remote: Compressing objects:  44% (723/1642)        
remote: Compressing objects:  45% (739/1642)        
remote: Compressing objects:  46% (756/1642)        
remote: Compressing objects:  47% (772/1642)        
remote: Compressing objects:  48% (789/1642)        
remote: Compressing objects:  49% (805/1642)        
remote: Compressing objects:  50% (821/1642)        
remote: Compressing objects:  51% (838/1642)        
remote: Compressing objects:  52% (854/1642)        
remote: Compressing objects:  53% (871/1642)        
remote: Compressing objects:  54% (887/1642)        
remote: Compressing objects:  55% (904/1642)        
remote: Compressing objects:  56% (920/1642)        
remote: Compressing objects:  57% (936/1642)        
remote: Compressing objects:  58% (953/1642)        
remote: Compressing objects:  59% (969/1642)        
remote: Compressing objects:  60% (986/1642)        
remote: Compressing objects:  61% (1002/1642)        
remote: Compressing objects:  62% (1019/1642)        
remote: Compressing objects:  63% (1035/1642)        
remote: Compressing objects:  64% (1051/1642)        
remote: Compressing objects:  65% (1068/1642)        
remote: Compressing objects:  66% (1084/1642)        
remote: Compressing objects:  67% (1101/1642)        
remote: Compressing objects:  68% (1117/1642)        
remote: Compressing objects:  69% (1133/1642)        
remote: Compressing objects:  70% (1150/1642)        
remote: Compressing objects:  71% (1166/1642)        
remote: Compressing objects:  72% (1183/1642)        
remote: Compressing objects:  73% (1199/1642)        
remote: Compressing objects:  74% (1216/1642)        
remote: Compressing objects:  75% (1232/1642)        
remote: Compressing objects:  76% (1248/1642)        
remote: Compressing objects:  77% (1265/1642)        
remote: Compressing objects:  78% (1281/1642)        
remote: Compressing objects:  79% (1298/1642)        
remote: Compressing objects:  80% (1314/1642)        
remote: Compressing objects:  81% (1331/1642)        
remote: Compressing objects:  82% (1347/1642)        
remote: Compressing objects:  83% (1363/1642)        
remote: Compressing objects:  84% (1380/1642)        
remote: Compressing objects:  85% (1396/1642)        
remote: Compressing objects:  86% (1413/1642)        
remote: Compressing objects:  87% (1429/1642)        
remote: Compressing objects:  88% (1445/1642)        
remote: Compressing objects:  89% (1462/1642)        
remote: Compressing objects:  90% (1478/1642)        
remote: Compressing objects:  91% (1495/1642)        
remote: Compressing objects:  92% (1511/1642)        
remote: Compressing objects:  93% (1528/1642)        
remote: Compressing objects:  94% (1544/1642)        
remote: Compressing objects:  95% (1560/1642)        
remote: Compressing objects:  96% (1577/1642)        
remote: Compressing objects:  97% (1593/1642)        
remote: Compressing objects:  98% (1610/1642)        
remote: Compressing objects:  99% (1626/1642)        
remote: Compressing objects: 100% (1642/1642)        
remote: Compressing objects: 100% (1642/1642), done.        
Receiving objects:   0% (1/40851)
Receiving objects:   0% (364/40851), 220.01 KiB | 207.00 KiB/s
Receiving objects:   1% (409/40851), 220.01 KiB | 207.00 KiB/s
Receiving objects:   2% (818/40851), 324.01 KiB | 204.00 KiB/s
Receiving objects:   2% (888/40851), 324.01 KiB | 204.00 KiB/s
Receiving objects:   3% (1226/40851), 516.01 KiB | 193.00 KiB/s
Receiving objects:   3% (1441/40851), 516.01 KiB | 193.00 KiB/s
Receiving objects:   4% (1635/40851), 628.01 KiB | 196.00 KiB/s
Receiving objects:   5% (2043/40851), 724.01 KiB | 195.00 KiB/s
Receiving objects:   5% (2290/40851), 724.01 KiB | 195.00 KiB/s
Receiving objects:   6% (2452/40851), 724.01 KiB | 195.00 KiB/s
Receiving objects:   7% (2860/40851), 852.01 KiB | 199.00 KiB/s
Receiving objects:   7% (3049/40851), 956.01 KiB | 192.00 KiB/s
Receiving objects:   8% (3269/40851), 988.01 KiB | 177.00 KiB/s
Receiving objects:   8% (3506/40851), 1.04 MiB | 169.00 KiB/s  
Receiving objects:   9% (3677/40851), 1.04 MiB | 169.00 KiB/s
Receiving objects:  10% (4086/40851), 1.18 MiB | 176.00 KiB/s
Receiving objects:  10% (4356/40851), 1.18 MiB | 176.00 KiB/s
Receiving objects:  11% (4494/40851), 1.26 MiB | 177.00 KiB/s
Receiving objects:  12% (4903/40851), 1.26 MiB | 177.00 KiB/s
Receiving objects:  13% (5311/40851), 1.39 MiB | 182.00 KiB/s
Receiving objects:  13% (5328/40851), 1.39 MiB | 182.00 KiB/s
Receiving objects:  13% (5470/40851), 1.64 MiB | 154.00 KiB/s
Receiving objects:  13% (5541/40851), 1.64 MiB | 154.00 KiB/s
Receiving objects:  13% (5598/40851), 1.76 MiB | 145.00 KiB/s
Receiving objects:  13% (5656/40851), 1.80 MiB | 129.00 KiB/s
Receiving objects:  13% (5702/40851), 1.86 MiB | 96.00 KiB/s 
Receiving objects:  14% (5720/40851), 1.86 MiB | 96.00 KiB/s
Receiving objects:  14% (5815/40851), 1.96 MiB | 92.00 KiB/s
Receiving objects:  14% (5986/40851), 2.15 MiB | 84.00 KiB/s
Receiving objects:  15% (6128/40851), 2.24 MiB | 103.00 KiB/s
Receiving objects:  15% (6165/40851), 2.36 MiB | 113.00 KiB/s
Receiving objects:  15% (6373/40851), 2.47 MiB | 135.00 KiB/s
Receiving objects:  16% (6537/40851), 2.67 MiB | 173.00 KiB/s
Receiving objects:  16% (6740/40851), 2.67 MiB | 173.00 KiB/s
Receiving objects:  17% (6945/40851), 2.80 MiB | 205.00 KiB/s
Receiving objects:  17% (7211/40851), 2.91 MiB | 206.00 KiB/s
Receiving objects:  17% (7254/40851), 2.96 MiB | 175.00 KiB/s
Receiving objects:  17% (7343/40851), 3.04 MiB | 157.00 KiB/s
Receiving objects:  18% (7354/40851), 3.04 MiB | 157.00 KiB/s
Receiving objects:  18% (7405/40851), 3.15 MiB | 126.00 KiB/s
Receiving objects:  18% (7471/40851), 3.21 MiB | 95.00 KiB/s 
Receiving objects:  18% (7505/40851), 3.24 MiB | 77.00 KiB/s
Receiving objects:  19% (7762/40851), 3.29 MiB | 67.00 KiB/s
Receiving objects:  19% (8068/40851), 3.39 MiB | 82.00 KiB/s
Receiving objects:  20% (8171/40851), 3.39 MiB | 82.00 KiB/s
Receiving objects:  21% (8579/40851), 3.39 MiB | 82.00 KiB/s
Receiving objects:  22% (8988/40851), 3.50 MiB | 93.00 KiB/s
Receiving objects:  22% (9176/40851), 3.65 MiB | 119.00 KiB/s
Receiving objects:  23% (9396/40851), 3.71 MiB | 120.00 KiB/s
Receiving objects:  23% (9447/40851), 3.80 MiB | 133.00 KiB/s
Receiving objects:  24% (9805/40851), 3.80 MiB | 133.00 KiB/s
Receiving objects:  24% (10035/40851), 3.88 MiB | 145.00 KiB/s
Receiving objects:  25% (10213/40851), 3.93 MiB | 152.00 KiB/s
Receiving objects:  26% (10622/40851), 3.93 MiB | 152.00 KiB/s
Receiving objects:  27% (11030/40851), 4.00 MiB | 164.00 KiB/s
Receiving objects:  27% (11375/40851), 4.00 MiB | 164.00 KiB/s
Receiving objects:  28% (11439/40851), 4.00 MiB | 164.00 KiB/s
Receiving objects:  29% (11847/40851), 4.09 MiB | 164.00 KiB/s
Receiving objects:  30% (12256/40851), 4.09 MiB | 164.00 KiB/s
Receiving objects:  31% (12664/40851), 4.17 MiB | 161.00 KiB/s
Receiving objects:  31% (12680/40851), 4.17 MiB | 161.00 KiB/s
Receiving objects:  31% (13045/40851), 4.23 MiB | 111.00 KiB/s
Receiving objects:  32% (13073/40851), 4.23 MiB | 111.00 KiB/s
Receiving objects:  33% (13481/40851), 4.27 MiB | 106.00 KiB/s
Receiving objects:  33% (13581/40851), 4.29 MiB | 91.00 KiB/s 
Receiving objects:  33% (13720/40851), 4.31 MiB | 82.00 KiB/s
Receiving objects:  34% (13890/40851), 4.31 MiB | 82.00 KiB/s
Receiving objects:  34% (14148/40851), 4.37 MiB | 66.00 KiB/s
Receiving objects:  35% (14298/40851), 4.37 MiB | 66.00 KiB/s
Receiving objects:  36% (14707/40851), 4.37 MiB | 66.00 KiB/s
Receiving objects:  37% (15115/40851), 4.50 MiB | 75.00 KiB/s
Receiving objects:  37% (15137/40851), 4.50 MiB | 75.00 KiB/s
Receiving objects:  38% (15524/40851), 4.50 MiB | 75.00 KiB/s
Receiving objects:  39% (15932/40851), 4.66 MiB | 91.00 KiB/s
Receiving objects:  39% (16195/40851), 4.79 MiB | 111.00 KiB/s
Receiving objects:  40% (16341/40851), 4.88 MiB | 129.00 KiB/s
Receiving objects:  40% (16733/40851), 5.04 MiB | 155.00 KiB/s
Receiving objects:  41% (16749/40851), 5.04 MiB | 155.00 KiB/s
Receiving objects:  42% (17158/40851), 5.15 MiB | 175.00 KiB/s
Receiving objects:  42% (17206/40851), 5.27 MiB | 197.00 KiB/s
Receiving objects:  42% (17416/40851), 5.36 MiB | 212.00 KiB/s
Receiving objects:  43% (17566/40851), 5.46 MiB | 238.00 KiB/s
Receiving objects:  44% (17975/40851), 5.60 MiB | 241.00 KiB/s
Receiving objects:  44% (17989/40851), 5.78 MiB | 244.00 KiB/s
Receiving objects:  44% (18380/40851), 5.91 MiB | 229.00 KiB/s
Receiving objects:  45% (18383/40851), 5.91 MiB | 229.00 KiB/s
Receiving objects:  45% (18398/40851), 5.97 MiB | 176.00 KiB/s
Receiving objects:  45% (18462/40851), 6.00 MiB | 160.00 KiB/s
Receiving objects:  45% (18503/40851), 6.03 MiB | 135.00 KiB/s
Receiving objects:  45% (18539/40851), 6.08 MiB | 105.00 KiB/s
Receiving objects:  45% (18701/40851), 6.25 MiB | 81.00 KiB/s 
Receiving objects:  46% (18792/40851), 6.25 MiB | 81.00 KiB/s
Receiving objects:  46% (18964/40851), 6.36 MiB | 80.00 KiB/s
Receiving objects:  46% (19013/40851), 6.60 MiB | 116.00 KiB/s
Receiving objects:  46% (19116/40851), 6.71 MiB | 139.00 KiB/s
Receiving objects:  46% (19183/40851), 6.89 MiB | 174.00 KiB/s
Receiving objects:  47% (19200/40851), 6.98 MiB | 182.00 KiB/s
Receiving objects:  47% (19220/40851), 7.00 MiB | 160.00 KiB/s
Receiving objects:  47% (19244/40851), 7.04 MiB | 114.00 KiB/s
Receiving objects:  47% (19254/40851), 7.07 MiB | 100.00 KiB/s
Receiving objects:  47% (19256/40851), 7.10 MiB | 74.00 KiB/s 
Receiving objects:  47% (19266/40851), 7.14 MiB | 46.00 KiB/s
Receiving objects:  47% (19338/40851), 7.23 MiB | 45.00 KiB/s
Receiving objects:  47% (19437/40851), 7.36 MiB | 64.00 KiB/s
Receiving objects:  48% (19609/40851), 7.45 MiB | 75.00 KiB/s
Receiving objects:  48% (19622/40851), 7.45 MiB | 75.00 KiB/s
Receiving objects:  48% (19913/40851), 7.64 MiB | 108.00 KiB/s
Receiving objects:  49% (20017/40851), 7.64 MiB | 108.00 KiB/s
Receiving objects:  49% (20329/40851), 7.94 MiB | 174.00 KiB/s
Receiving objects:  49% (20370/40851), 8.18 MiB | 196.00 KiB/s
Receiving objects:  49% (20408/40851), 8.22 MiB | 172.00 KiB/s
Receiving objects:  49% (20408/40851), 8.26 MiB | 142.00 KiB/s
Receiving objects:  49% (20408/40851), 8.30 MiB | 95.00 KiB/s 
Receiving objects:  49% (20408/40851), 8.34 MiB | 72.00 KiB/s
Receiving objects:  50% (20426/40851), 8.34 MiB | 72.00 KiB/s
Receiving objects:  50% (20502/40851), 8.42 MiB | 58.00 KiB/s
Receiving objects:  50% (20631/40851), 8.59 MiB | 74.00 KiB/s
Receiving objects:  50% (20730/40851), 8.76 MiB | 99.00 KiB/s
Receiving objects:  50% (20752/40851), 8.80 MiB | 103.00 KiB/s
Receiving objects:  51% (20835/40851), 8.85 MiB | 110.00 KiB/s
Receiving objects:  51% (20853/40851), 8.91 MiB | 119.00 KiB/s
Receiving objects:  51% (21038/40851), 9.11 MiB | 126.00 KiB/s
Receiving objects:  51% (21079/40851), 9.14 MiB | 113.00 KiB/s
Receiving objects:  51% (21111/40851), 9.18 MiB | 87.00 KiB/s 
Receiving objects:  51% (21185/40851), 9.20 MiB | 75.00 KiB/s
Receiving objects:  51% (21206/40851), 9.24 MiB | 62.00 KiB/s
Receiving objects:  52% (21243/40851), 9.24 MiB | 62.00 KiB/s
Receiving objects:  52% (21292/40851), 9.24 MiB | 62.00 KiB/s
Receiving objects:  52% (21426/40851), 9.43 MiB | 65.00 KiB/s
Receiving objects:  53% (21652/40851), 9.54 MiB | 74.00 KiB/s
Receiving objects:  53% (21902/40851), 9.63 MiB | 87.00 KiB/s
Receiving objects:  54% (22060/40851), 9.76 MiB | 110.00 KiB/s
Receiving objects:  54% (22272/40851), 9.89 MiB | 135.00 KiB/s
Receiving objects:  55% (22469/40851), 10.12 MiB | 178.00 KiB/s
Receiving objects:  55% (22519/40851), 10.12 MiB | 178.00 KiB/s
Receiving objects:  55% (22741/40851), 10.34 MiB | 202.00 KiB/s
Receiving objects:  55% (22843/40851), 10.36 MiB | 186.00 KiB/s
Receiving objects:  56% (22877/40851), 10.39 MiB | 167.00 KiB/s
Receiving objects:  56% (22940/40851), 10.42 MiB | 150.00 KiB/s
Receiving objects:  56% (22972/40851), 10.43 MiB | 126.00 KiB/s
Receiving objects:  56% (23030/40851), 10.52 MiB | 92.00 KiB/s 
Receiving objects:  56% (23192/40851), 10.66 MiB | 75.00 KiB/s
Receiving objects:  57% (23286/40851), 10.66 MiB | 75.00 KiB/s
Receiving objects:  57% (23347/40851), 10.74 MiB | 75.00 KiB/s
Receiving objects:  57% (23663/40851), 10.91 MiB | 102.00 KiB/s
Receiving objects:  58% (23694/40851), 10.91 MiB | 102.00 KiB/s
Receiving objects:  58% (23989/40851), 11.19 MiB | 155.00 KiB/s
Receiving objects:  59% (24103/40851), 11.37 MiB | 189.00 KiB/s
Receiving objects:  59% (24361/40851), 11.50 MiB | 209.00 KiB/s
Receiving objects:  60% (24511/40851), 11.50 MiB | 209.00 KiB/s
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 8191 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output

	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2846)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:2185)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:635)
	at jenkins.plugins.git.AbstractGitSCMSource$8.run(AbstractGitSCMSource.java:605)
	at jenkins.plugins.git.AbstractGitSCMSource$8.run(AbstractGitSCMSource.java:591)
	at jenkins.plugins.git.AbstractGitSCMSource.doRetrieve(AbstractGitSCMSource.java:397)
	at jenkins.plugins.git.AbstractGitSCMSource.doRetrieve(AbstractGitSCMSource.java:351)
	at jenkins.plugins.git.AbstractGitSCMSource.retrieve(AbstractGitSCMSource.java:591)
	at jenkins.scm.api.SCMSource._retrieve(SCMSource.java:372)
	at jenkins.scm.api.SCMSource.fetch(SCMSource.java:282)
	at jenkins.branch.MultiBranchProject.computeChildren(MultiBranchProject.java:654)
	at com.cloudbees.hudson.plugins.folder.computed.ComputedFolder.updateChildren(ComputedFolder.java:269)
	at com.cloudbees.hudson.plugins.folder.computed.FolderComputation.run(FolderComputation.java:167)
	at jenkins.branch.MultiBranchProject$BranchIndexing.run(MultiBranchProject.java:1057)
	at hudson.model.ResourceController.execute(ResourceController.java:101)
	at hudson.model.Executor.run(Executor.java:442)
Finished: FAILURE
```

**Comment by dd010101** [comment]
Looks like network/server/overload temporary issue. Not much to do - you can just retry the build again. Find those failed jobs and run them manually in Jenkins via **Build now** and it should complete eventually.

**Comment by phila6** [comment]
Yeah I have tried it a multiple times(Build Now) and it failed. 
Is it possible to run this step manually, while still using the scripts(which are super helpful)? Or have to start from beginning using  manual process?

**Comment by dd010101** [comment]
> Yeah I have tried it a multiple times(Build Now) and it failed.

The same way ? I would expect this would go away after while. Can you `git clone https://github.com/vyos/vyos-cloud-init.git` and then `git fetch --all` outside the build, just on the same machine?

> Is it possible to run this step manually, while still using the scripts(which are super helpful)? Or have to start from beginning using manual process?

The manual process would yield same result since the failing script is the Jenkins job = build script of VyOS - not the automated script. The automated scripts prepare environment and populate the Jenkins jobs with official build scripts of VyOS (the same way as the manual process). This doesn't look like bug of the build script though - seems like connection issue to the github.com? This is the only failing package or are there also others? Identical fetch is used by nearly all jobs/packages.

**Comment by phila6** [comment]
I tried cloning it, it times out with same error message. 
When you look at the git progress indicator, it starts up fine, and gets to 30-40% quickly, but then slows down, not sure if maybe some throttling kicks in.

I was able to git clone with a "--depth=1" option however.

**Comment by dd010101** [comment]
Yep, looks like the traffic is somehow identified as suspicious. This doesn't look like anything we can fix but perhaps you can find workaround.

If this is the only case then you could modify the Jenkins job to do more smaller clone but I don't know if this isn't spread to all packages then different solution would be required.

If you find the **vyos-cloud-init** job (the base job/pipeline without specific branch) then you have option to **Configure** in the navigation. There you can unstick the **Tags** under **Advanced clone behaviours** save and try build again, and if this will not work then in the same section try to tick **Shallow clone**.

If the issue was only with cloud init then perhaps this would work as workaround but we can't have this setting for all packages, some packages do need full fetch since they are using the metadata from repository in the build. 

I did have similar issues in past where many connections at once would trigger some protection mechanism and very good solution was to tunnel the traffic so the firewall or whatever is limiting the traffic doesn't sees it. I had my own VPS so I use it as gateway and tunneled the many connections traffic via VPN to it. Perhaps this would be something what would solve these issues completely if you would somehow tunnel the traffic of the build OS to somewhere else so it doesn't flow naked.

As easier option you could also try to add some traffic limiter like shaper/tc - to add latency, this would make everything including the clone less aggressive. You can try to start with 10ms and then increase the latency via tc qdisk on your interface of the build OS until it starts working.

```
tc qdisc add dev <INTERFACE> root netem delay 10ms
```

**Comment by marekm72** [comment]
I could reproduce the issue as well - on a slow connection, WireGuard tunnel over a mobile Internet connection (macOS 10.13.6 connected via WiFi to iPhone personal hotspot sharing LTE connection)   But it works if I use "git clone --depth=1", works over the same mobile connection without a tunnel, works over the fast connection from my server (which I access remotely over the slow connection as I'm far away right now), and works over the same slow connection (with or without WireGuard tunnel) from my mirror - git clone https://git.amelek.net/vyos-cloud-init.git - so it looks like a weird MTU issue, probably at the GitHub end.

**Comment by dd010101** [comment]
I can't reproduce this. I did try to clone vyos-cloud-init many times under different conditions - for example I did limit my bandwidth to crawl, then it taken couple minutes to do the clone of vyos-cloud-init but it ended successfully. I also rebuild the Jenkins+all packages many times and I didn't see such issue with any repositories and there are even bigger ones than vyos-cloud-init.

If it smells like MTU then you can set local MSS clamp to low value and see if that helps? Something like:

```
iptables -I OUTPUT -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --set-mss 1200
```

**Comment by marekm72** [comment]
I've tried on the router that terminates my WG tunnel (RB5009 running ROS 7.15.2):
```
/ip/firewall/mangle
add action=change-mss chain=forward new-mss=1200 passthrough=yes protocol=tcp tcp-flags=syn
```
but it made no difference.
No NAT here, I actually tunnel a public IPv4 to the laptop (and IPv6 too but it seems broken on this old macOS, the same WG setup works fine on a newer iPhone), so it's really weird.  My mirror doesn't fail even when the mobile connection is heavily loaded by some large download.  The largest clone depth that works with GH is about 350.

@phila6 does `git clone https://git.amelek.net/vyos-cloud-init.git` work for you?

**Comment by phila6** [comment]
I did more testing based on the suggestions, here is what I see:

1. The only 2 repos I see the issues with are vyos-1x and vyos-cloud-init
2. Disabling Tags and/or enabling Shallow clone in Jenkins does not change things, still fails
3. Increasing latency via qdisk did not change things, I tried scaling up to 60ms
4. MSS clamping did not help
5. Doing a packet capture for the Github traffic, I see a lot of TCP retransmits. Nothing related to MTU, or window not scaling up due to latency. 
6. Manually downloading a zip file of the repo works, though TCP retransmits are still present, it completes the download eventually
7. Downloading something from kernel.org, is also slow and there are retransmits.

From this I think there is something causing packet loss, probably related to my VM setup(just a Qemu Debian VM).

I dont think this is related to scripts, or Github itself.

Will go troubleshoot my setup. Appreciate everybody looking at this.

**Comment by phila6** [comment]
Another update. TLDR - it was due to my VM network config.


My build debian VM, was configured under GNS3, and connection to real network was done through a Cloud type of node, this allows your VM to have an IP on local LAN, so I can have access to Jenkins UI.
But using Cloud object, was introducing packet loss, after I switched to connect the VM through a NAT object, it works fine. I lost access to Jenkins console but I guess it is ok, as long as it does its job.  
On step 7 now, building the packages.

Thank you all for assistance.

**Comment by marekm72** [comment]
Glad you sorted it out. Still weird it also happens to me accidentally in a completely different setup, and much larger downloads (like ~130MB wireshark update) work fine at the same time over the same rather poor connection, just some GH repos fail reproducibly every time. TCP is designed to handle some packet loss (common on congested networks) gracefully, it becomes slow but it takes much larger loss and very long timeouts to completely break the connection, while here it goes slowly but makes some progress and then suddenly fails. It could actually be something on the GH end, nothing of this complexity is 100% bug-free. Might be good to try to make the simplest possible setup that reproduces this and send a bug report to GH.

**Comment by dd010101** [comment]
In my experience such funny behavior is often caused by IDS/IPS system detecting "suspicious" pattern. That's why I suggested tunneling but if the misbehaving IDS/IPS system is on server side and not your provider side then you are out of luck.

It seems like the common trigger is lossy link / heavy packet drop. So I did simulate bad link by tc:

```
tc qdisc add dev enp1s0 root netem delay 100ms 10ms 75% loss 50% 75%
```
100ms+-10ms with 75% of chance
+50% packet loss with 75% of chance

Then it breaks but I did really need completely crazy loss to make it fail. Slight or even heavy loss is working fine so I'm not sure if this is feature or bug. Like 25% loss is bad and that works just fine so I don't think it's unreasonable to drop 50% lossy connections since then the people should fix their link and not bother the server with so many retries 😄.

Anyway I think the original issue is resolved and the cause is known - bad link causing GitHub servers to give up on the connection if you do big transfers.
